cedar-policy-core/src/entities/json/schema_types.rs

```diff
@@ -382,9 +379,8 @@ pub fn schematype_of_restricted_expr(
                         BorrowedRestrictedExpr::new_unchecked(v), // assuming the invariant holds for the record as a whole, it will also hold for each attribute value
                         extensions,
                     )?;
-                    // we can't know if the attribute is required or optional,
-                    // but marking it optional is more flexible -- allows the
-                    // attribute type to `is_consistent_with()` more types
+                    // We can't know if the attribute is required or optional.
+                    // Keep as optional to minimize heterogeneous sets.
```

Comment:
How relevant is this comment with the latest changes?

Reply:
We investigated this and decided we'd prefer the behavior of `is_consistent_with` with this set to `required`. So I replaced what I view as a misleading comment with the reason we didn't proceed down that route.